core/actionProxy/actionproxy.py

```diff
@@ -136,13 +136,17 @@ def error(msg):
         # run the process and wait until it completes.
         (o, e) = p.communicate()
 
+        # stdout/stderr may be either text or bytes, depending on Python
+        # version.   In the latter case, decode to text.
+        o, e = map(lambda x: x.decode() if isinstance(x, bytes) else x, (o, e))
```

Comment:
> The map(lambda) makes my head spin.

Fair enough.

> Also line 143 (if o is not None) will always be True. o might be '' but it will never be None. Same for e on line 151.

True.  The check was already there in existing code.  I did not remove it originally because it might be required if someone were to change the parameters passed to `Popen`.  It felt safer to leave it.

> Finally, I would not recommend all of the u'' changes unless you see an absolute necessity.

Removed. Not necessary, just the best practice mentioned in the above comment.